scripts/dashboard/install-cephadm-e2e-deps.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+
+set -ex
+
+on_error() {
+    if [ "$1" != "0" ]; then
+        printf "\n\nERROR $1 thrown on line $2\n\n"
+        printf "\n\nCollecting info...\n\n"
+        sudo journalctl --since "10 min ago" --no-tail --no-pager -x
+        printf "\n\nERROR: displaying containers' logs:\n\n"
+        docker ps -aq | xargs docker logs
+        printf "\n\nTEST FAILED.\n\n"
+    fi
+}
+
+trap 'on_error $? $LINENO' ERR
+
+sudo apt -y install libvirt-daemon-system libvirt-daemon-driver-qemu qemu-kvm libvirt-clients
+
+sudo usermod -aG libvirt $(id -un)
+newgrp libvirt  # Avoid having to log out and log in for group addition to take effect.
+sudo systemctl enable --now libvirtd
+
+if [[ $(command -v docker) == '' ]]; then
+    # Set up docker official repo and install docker.
+    sudo apt update -y
+    sudo apt install \
+    apt-transport-https \
+    ca-certificates \
+    curl \
+    gnupg \
+    lsb-release
+    curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
+    echo \
+        "deb [arch=amd64 signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu \
+        $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+    sudo apt update -y
+    sudo apt install -y docker-ce docker-ce-cli containerd.io
+fi
+sudo groupadd docker || true
+sudo usermod -aG docker $(id -un)
+sudo systemctl start docker
+sudo chgrp "$(id -un)" /var/run/docker.sock
+
+docker info
+docker container prune -f
+
+KCLI_CONFIG_DIR="${HOME}/.kcli"
+mkdir -p ${KCLI_CONFIG_DIR}
+if [[ ! -f "${KCLI_CONFIG_DIR}/id_rsa" ]]; then
+    ssh-keygen -t rsa -q -f "${KCLI_CONFIG_DIR}/id_rsa" -N ""
+fi
+
+: ${KCLI_CONTAINER_IMAGE:='quay.io/karmab/kcli'}
+
+echo "#!/usr/bin/env bash
+
+docker run --net host --security-opt label=disable \
+    -v ${KCLI_CONFIG_DIR}:/root/.kcli \
+    -v ${PWD}:/workdir \
+    -v /var/lib/libvirt/images:/var/lib/libvirt/images \
+    -v /var/run/libvirt:/var/run/libvirt \
+    -v /var/tmp:/ignitiondir \
+    ${KCLI_CONTAINER_IMAGE} \""'${@}'"\"
+" | sudo tee /usr/local/bin/kcli
+sudo chmod +x /usr/local/bin/kcli
+
+# Install required deps.
+sudo apt update -y
+sudo apt install -y nodejs npm openssh-server
+
+sudo mkdir -p /var/lib/libvirt/images/ceph-dashboard
+kcli create pool -p /var/lib/libvirt/images/ceph-dashboard ceph-dashboard
+kcli create network -c 192.168.100.0/24 ceph-dashboard
```

Comment:
Can you add a teardown script to clean this up after the job completes?

Reply:
@djgalloway The cleanup is tirggered when the job finishes on success/error here (in the other PR):
https://github.com/ceph/ceph/pull/42263/files?file-filters%5B%5D=.sh#diff-ccaf2a6af4475351e29cd34ed246c9fbd3cbbc14c93d961a360c283eecf02e88R5

In our last conversation we agreed to not erase the libvirt folder/pool/network but to rename it (s/default/ceph-dashboard) in order to not constantly download the VM fedora image and to avoid name conflicts in the future.
Do you prefer to completely erase it?

Reply:
Ah, right.  Fair enough.  Can you just put a comment in the file pointing to that cleanup function please?  The canonical link to the file in `master` is fine even though that PR isn't merged yet.

Reply:
Sure.

Reply:
Done.